clang/lib/CodeGen/CGExprAgg.cpp

```diff
@@ -537,8 +537,9 @@ void AggExprEmitter::EmitArrayInit(Address DestPtr, llvm::ArrayType *AType,
       elementType.isTriviallyCopyableType(CGF.getContext())) {
     CodeGen::CodeGenModule &CGM = CGF.CGM;
     ConstantEmitter Emitter(CGF);
+    Qualifiers Quals;
     QualType GVArrayQTy = CGM.getContext().getAddrSpaceQualType(
-        CGM.getContext().removeAddrSpaceQualType(ArrayQTy),
+        CGM.getContext().getUnqualifiedArrayType(ArrayQTy, Quals),
```

Comment:
I see two issues with this fix:

1. It leaves removeAddrSpaceQualType broken for anyone else who tries to use it.  Ideally, removeAddrSpaceQualType should do what it says.
2. This drops other qualifiers on the floor; they might be relevant in some cases.

Reply:
Do you mean we should actually fix removeAddrSpaceQualType? Somewhere inside removeAddrSpaceQualType, we
should use getUnqualifiedArrayType if it is an arrayType, and getSingleStepDesugaredType othereise?
I have to admit that I have no experience in this field, so I am relying on you and @svenvh to move on for a reasonable fix. Thanks.

Reply:
Something like that, yes.

Reply:
> Something like that, yes.

Thanks. Updated. Should be closer!